nemo/collections/asr/models/label_models.py

```diff
@@ -87,39 +86,62 @@ def list_available_models(cls) -> List[PretrainedModelInfo]:
 
     def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.world_size = 1
-        self.cal_labels_occurrence = False
+        self.cal_labels_occurrence_train = False
+        self.labels_occurrence = None
+
+        if 'loss' in cfg:
+            if 'weight' in cfg.loss:
+                if cfg.loss.weight == 'auto':
+                    self.cal_labels_occurrence_train = True
+                else:
+                    weight = cfg.loss.weight
+            else:
+                weight = None  # weight is None for angular loss and CE loss if it's not specified.
+
         if trainer is not None:
             self.world_size = trainer.num_nodes * trainer.num_devices
 
         super().__init__(cfg=cfg, trainer=trainer)
 
-        self.preprocessor = EncDecSpeakerLabelModel.from_config_dict(cfg.preprocessor)
-        self.encoder = EncDecSpeakerLabelModel.from_config_dict(cfg.encoder)
-        self.decoder = EncDecSpeakerLabelModel.from_config_dict(cfg.decoder)
+        if self.labels_occurrence:
+            # Goal is to give more weight to the classes with less samples so as to match the ones with the higher frequencies
+            weight = [sum(self.labels_occurrence) / (len(self.labels_occurrence) * i) for i in self.labels_occurrence]
 
         if 'loss' in cfg:
-            if 'angular' in cfg.decoder and cfg.decoder['angular']:
-                logging.info("loss is Angular Softmax")
-                scale = cfg.loss.scale
-                margin = cfg.loss.margin
-                self.loss = AngularSoftmaxLoss(scale=scale, margin=margin)
-            else:
-                logging.info("loss is Softmax-CrossEntropy")
-                if 'weight' in cfg.loss and cfg.loss['weight']:
-                    if cfg.loss.weight == 'auto':
-                        self.cal_labels_occurrence = True
-                        # Goal is to give more weight to the classes with less samples so as to match the ones with the higher frequencies
-                        weight = [
-                            sum(self.labels_occurrence) / (len(self.labels_occurrence) * i)
-                            for i in self.labels_occurrence
-                        ]
+            # To support older version checkpoints
+            if '_target_' not in cfg.loss:
+                logging.info("Support older version checkpoints of label models")
```

Comment:
instead of this logging message, can you add:
setting angular=False in decoder is deprecated and will be removed in 1.13 version, use specific loss with `_target_`

Reply:
Sure